src/main/java/com/scalar/cassy/transferer/BackupPath.java

```diff
@@ -11,20 +11,29 @@
   private static final String CLUSTER_BACKUP_KEY = "cluster-backup";
   private static final String NODE_BACKUP_KEY = "node-backup";
 
-  public static String create(BackupConfig config, String path) {
-    String type = NODE_BACKUP_KEY;
-    if (config.getBackupType().equals(BackupType.CLUSTER_SNAPSHOT)) {
-      type = CLUSTER_BACKUP_KEY;
-    }
-    return create(type, config, path);
+  public static String create(BaseConfig config, String path) {
+    return create(getType(config), config, path);
   }
 
-  public static String create(RestoreConfig config, String path) {
-    String type = NODE_BACKUP_KEY;
-    if (config.getRestoreType().equals(RestoreType.CLUSTER)) {
-      type = CLUSTER_BACKUP_KEY;
+  public static String getType(BaseConfig config) {
+    if (config instanceof RestoreConfig) {
```

Comment:
I think it can be simplified something like this. (the format will be changed by google format)
```
public static String getType(BaseConfig config) {
    String type = NODE_BACKUP_KEY;
    if ((config instanceof RestoreConfig && 
         restoreConfig.getRestoreType().equals(RestoreType.CLUSTER)) ||
        (config instanceof BackupConfig &&
         backupConfig.getBackupType().equals(BackupType.CLUSTER_SNAPSHOT))) {
       type = CLUSTER_BACKUP_KEY;
    }
    return type;
}
```

Reply:
@Torch3333 Please take a look when you get a chance.

Reply:
Sorry for the delay, I updated it.